plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -275,11 +286,41 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred opening
-	conn, err := sql.Open("mssql", server)
-	if err != nil {
-		return err
+	// connection object
+	var conn *sql.DB
```

Comment:
Heads up: Depending on which [PR ](https://github.com/influxdata/telegraf/pull/8596/files)gets merged to master, there  may be a possibility of merge conflict

Reply:
thank you, yes I did notice that PR